src/neo/SmartContract/Native/Tokens/NeoToken.Governance.cs

```diff
@@ -0,0 +1,186 @@
+            if (state_account.VoteTo != null)
+            {
+                StorageKey key = CreateStorageKey(Prefix_Candidate).Add(state_account.VoteTo);
+                StorageItem storage_validator = snapshot.Storages.GetAndChange(key);
+                CandidateState state_validator = storage_validator.GetInteroperable<CandidateState>();
+                state_validator.Votes -= state_account.Balance;
+                if (!state_validator.Registered && state_validator.Votes.IsZero)
+                    snapshot.Storages.Delete(key);
+            }
+            state_account.VoteTo = voteTo;
+            if (voteTo != null)
+            {
+                StorageKey key = CreateStorageKey(Prefix_Candidate).Add(voteTo);
+                if (snapshot.Storages.TryGet(key) is null) return false;
+                StorageItem storage_validator = snapshot.Storages.GetAndChange(key);
+                CandidateState state_validator = storage_validator.GetInteroperable<CandidateState>();
+                if (!state_validator.Registered) return false;
+                state_validator.Votes += state_account.Balance;
+            }
+            return true;
+        }
+
+        [ContractMethod(1_00000000, CallFlags.AllowStates)]
+        public (ECPoint PublicKey, BigInteger Votes)[] GetCandidates(StoreView snapshot)
+        {
+            return GetCandidatesInternal(snapshot).ToArray();
+        }
+
+        private IEnumerable<(ECPoint PublicKey, BigInteger Votes)> GetCandidatesInternal(StoreView snapshot)
+        {
+            byte[] prefix_key = StorageKey.CreateSearchPrefix(Id, new[] { Prefix_Candidate });
+            return snapshot.Storages.Find(prefix_key).Select(p =>
+            (
+                p.Key.Key.AsSerializable<ECPoint>(1),
+                p.Value.GetInteroperable<CandidateState>()
+            )).Where(p => p.Item2.Registered).Select(p => (p.Item1, p.Item2.Votes));
+        }
+
+        [ContractMethod(1_00000000, CallFlags.AllowStates)]
+        public ECPoint[] GetValidators(StoreView snapshot)
+        {
+            return GetCommitteeMembers(snapshot, ProtocolSettings.Default.MaxValidatorsCount).OrderBy(p => p).ToArray();
+        }
+
+        [ContractMethod(1_00000000, CallFlags.AllowStates)]
+        public ECPoint[] GetCommittee(StoreView snapshot)
+        {
+            return GetCommitteeMembers(snapshot, ProtocolSettings.Default.MaxCommitteeMembersCount).OrderBy(p => p).ToArray();
+        }
+
+        public UInt160 GetCommitteeAddress(StoreView snapshot)
+        {
+            ECPoint[] committees = GetCommittee(snapshot);
+            return Contract.CreateMultiSigRedeemScript(committees.Length - (committees.Length - 1) / 2, committees).ToScriptHash();
+        }
+
+        private IEnumerable<ECPoint> GetCommitteeMembers(StoreView snapshot, int count)
+        {
+            return GetCandidatesInternal(snapshot).OrderByDescending(p => p.Votes).ThenBy(p => p.PublicKey).Select(p => p.PublicKey).Take(count);
```

Comment:
```suggestion
            return GetCandidatesInternal(snapshot).OrderByDescending(p => p.Votes).ThenBy(p => p.PublicKey).Select(p => p.PublicKey).Take(count).OrderBy(p => p);
```

And remove the other order by?